pkg/codegen/importer/hcl2.go

```diff
@@ -425,7 +426,68 @@ func generateValue(typ schema.Type, value resource.PropertyValue) (model.Express
 					})
 				}
 			}
-		} else {
+		case *schema.UnionType:
+			discriminatorPropertyName := objectType.Discriminator
+			selectElementType := func() (*schema.ObjectType, bool) {
```

Comment:
Why do we create a lambda here, only to invoke it on the next line? We should either separate out `selectElementType` into its own function or inline it.

Reply:
Just a stylistic choice to make to more readable. I'll rewrite it.